### PR TITLE
Added ACME LDAP database config

### PR DESCRIPTION
**Pull request by edewata**
A sample `create.ldif` and `database.conf` has been added to initialize and configure an LDAP database for ACME.
    
The `LDAPDatabase` class has been modified to obtain the configuration parameters and password from `database.conf` in standalone ACME deployment scenario (without CA).

The `/usr/share/pki/server/conf/schema.ldif` has been modified such that it only contains only the original PKI schema. The ACME schema has been moved into `/usr/share/pki/acme/conf/database/ldap/schema.ldif`.

Updated docs:
- https://github.com/edewata/pki/blob/acme-dev/docs/installation/acme/Installing_ACME_Responder.md
- https://github.com/edewata/pki/blob/acme-dev/docs/installation/acme/Configuring_ACME_Database.md

### CODE REVIEW

**Comment by frasertweedale** [comment]
(Still didn't get to this - tomorrow I hope!)

File: docs/installation/acme/Configuring_ACME_Database.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+```
+$ ldapmodify -h $HOSTNAME -x -D "cn=Directory Manager" -w Secret.123 \
+    -f /usr/share/pki/acme/conf/database/ldap/schema.ldif
+```
+
+Next, create the ACME LDAP tree with the following command:
+
+```
+$ ldapadd -h $HOSTNAME -x -D "cn=Directory Manager" -w Secret.123 \
+    -f /usr/share/pki/acme/conf/database/ldap/create.ldif
```
[the remark this answers is not in the document]
Comment (edewata) [reply docs/installation/acme/Configuring_ACME_Database.md:42]: I updated the doc. Is this sufficient?

**Comment by edewata** [comment]
@frasertweedale No worries, take your time. Thanks!

File: base/server/src/com/netscape/cms/servlet/csadmin/LDAPConfigurator.java [hunk 2]
```diff
@@ -95,8 +95,11 @@ public void enableUSN() throws Exception {
     }
 
     public void setupSchema() throws Exception {
-        logger.info("Setting up schema");
+        logger.info("Setting up PKI schema");
         importSchemaFile("/usr/share/pki/server/conf/schema.ldif");
+
+        logger.info("Setting up ACME schema");
+        importSchemaFile("/usr/share/pki/acme/conf/database/ldap/schema.ldif");
```
Comment (frasertweedale) [inline base/server/src/com/netscape/cms/servlet/csadmin/LDAPConfigurator.java:102]: This is good.

As you have noted before in previous files, the ACME schema was defined twice:

- as a series of LDIF modify objects, in `...server/conf/schema.ldif` originally, and as of this PR in `...acme/conf/database/ldap/schema.ldif`.
- as an incomplete entry `cn=schema` entry in `...server/conf/schema-acme.ldif`.

I'll check whether the IPA schema update routines can handle LDIF modify objects, in which case we can delete the "plain object" definition.  Alternatively, we could modify `importSchemaFile` to interpret plain objects into modify operations, and remove the "modify" defintion (and move the "plain object" def to `...acme/conf/database/ldap/schema`).

Nothing to action here in this PR - just putting down my thoughts.

File: docs/installation/acme/Configuring_ACME_Database.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (frasertweedale) [reply docs/installation/acme/Configuring_ACME_Database.md:42]: Yeah it looks fine, thanks.

File: base/server/src/com/netscape/cms/servlet/csadmin/LDAPConfigurator.java (hunk 2 above)
Comment (frasertweedale) [reply base/server/src/com/netscape/cms/servlet/csadmin/LDAPConfigurator.java:102]: Confirmed, the IPA schema update machinery handles this file just fine.  I will make a separate PR later to remove the duplicate definition.

**Comment by frasertweedale** [comment]
Sorry for the delay.  It's good to go; merging now.

**Pull request merged**